pkg/noderesources/resource_allocation.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Most of this file was copied from the main k/k repo.
+// See: https://github.com/kubernetes/kubernetes/blob/v1.18.6/pkg/scheduler/framework/plugins/noderesources/resource_allocation.go
+
+package noderesources
+
+import (
+	v1 "k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/klog/v2"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"k8s.io/kubernetes/pkg/features"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	"k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	schedutil "k8s.io/kubernetes/pkg/scheduler/util"
+)
+
+// resourceToWeightMap contains resource name and weight.
+type resourceToWeightMap map[v1.ResourceName]int64
+
+// defaultResourcesToWeightMap is used to set default resourceToWeight map for CPU and memory.
+// The base unit for CPU is millicore, while the base using for memory is a byte.
+// The default CPU weight is 1<<20 and default memory weight is 1. That means a millicore
+// has a weighted score equivalent to 1 MiB.
+var defaultResourcesToWeightMap = resourceToWeightMap{v1.ResourceMemory: 1, v1.ResourceCPU: 1 << 20}
```

Comment:
Is it the only difference from the upstream version?

Reply:
Plus some other changes to make it work outside the scheduler/framework/plugins/noderesources package, and changes per a couple comments here (klog v1, allocable->allocatable). Diff with 1.19:

2c2
< Copyright 2017 The Kubernetes Authors.
---
> Copyright 2020 The Kubernetes Authors.
16a17,20
> // This file was copied from the main k/k repo and modified to work outside the scheduler/framework/plugins/noderesources pkg.
> // Only substantial difference is the addition of the var "defaultResourcesToWeightMap".
> // See: https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/scheduler/framework/plugins/noderesources/resource_allocation.go
> 
22c26
< 	"k8s.io/klog/v2"
---
> 	"k8s.io/klog"
25a30
> 	"k8s.io/kubernetes/pkg/scheduler/nodeinfo"
31a37,42
> // defaultResourcesToWeightMap is used to set default resourceToWeight map for CPU and memory.
> // The base unit for CPU is millicore, while the base using for memory is a byte.
> // The default CPU weight is 1<<20 and default memory weight is 1. That means a millicore
> // has a weighted score equivalent to 1 MiB.
> var defaultResourcesToWeightMap = resourceToWeightMap{v1.ResourceMemory: 1, v1.ResourceCPU: 1 << 20}
> 
38c49
< 	scorer              func(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64
---
> 	scorer              func(requested, allocatable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64
48c59
< 	nodeInfo *framework.NodeInfo) (int64, *framework.Status) {
---
> 	nodeInfo *nodeinfo.NodeInfo) (int64, *framework.Status) {
69c80
< 	if klog.V(10).Enabled() {
---
> 	if klog.V(10) {
92c103
< func calculateResourceAllocatableRequest(nodeInfo *framework.NodeInfo, pod *v1.Pod, resource v1.ResourceName) (int64, int64) {
---
> func calculateResourceAllocatableRequest(nodeInfo *nodeinfo.NodeInfo, pod *v1.Pod, resource v1.ResourceName) (int64, int64) {
96c107
< 		return nodeInfo.Allocatable.MilliCPU, (nodeInfo.NonZeroRequested.MilliCPU + podRequest)
---
> 		return nodeInfo.AllocatableResource().MilliCPU, (nodeInfo.NonZeroRequest().MilliCPU + podRequest)
98c109
< 		return nodeInfo.Allocatable.Memory, (nodeInfo.NonZeroRequested.Memory + podRequest)
---
> 		return nodeInfo.AllocatableResource().Memory, (nodeInfo.NonZeroRequest().Memory + podRequest)
101c112
< 		return nodeInfo.Allocatable.EphemeralStorage, (nodeInfo.Requested.EphemeralStorage + podRequest)
---
> 		return nodeInfo.AllocatableResource().EphemeralStorage, (nodeInfo.RequestedResource().EphemeralStorage + podRequest)
104c115
< 			return nodeInfo.Allocatable.ScalarResources[resource], (nodeInfo.Requested.ScalarResources[resource] + podRequest)
---
> 			return nodeInfo.AllocatableResource().ScalarResources[resource], (nodeInfo.RequestedResource().ScalarResources[resource] + podRequest)
107c118
< 	if klog.V(10).Enabled() {
---
> 	if klog.V(10) {